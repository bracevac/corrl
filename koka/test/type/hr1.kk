function xauto(f:forall<a> a -> a) {
  f(f)
}

function auto(f:forall<a> a -> a) : (forall<a> a -> a)
{
  f(f)
}

val test1 = auto(id)
val test2 = apply(auto,id)
val ids   = single(id:forall<a> a -> a)  //rigid annotation
val test3  = ids.map(auto)
val test3a = ids.map(xauto)

function test4()  -> apply(head,ids)
function test4a() -> apply(tail,ids)

function test5()  -> apply(fun(xs:list<list<forall<a> a -> a>>){ xs.map(head) },single(ids))

function test6()  -> choose(id,auto)
function test6a() -> choose(auto,id)
function test7()  -> choose(xauto,xauto)

function test8()  -> choose(ids,[])
function test8a() -> choose([],ids)
