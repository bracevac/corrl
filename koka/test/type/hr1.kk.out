auto : (f : forall<a> (a) -> a) -> (forall<a> (a) -> a)
ids : list<forall<a> (a) -> a>
test1 : forall<a> (a) -> a
test2 : forall<a> (a) -> a
test3 : list<forall<a> (a) -> a>
test3a : forall<a> list<(a) -> a>
test4 : () -> maybe<forall<a> (a) -> a>
test4a : () -> list<forall<a> (a) -> a>
test5 : () -> list<maybe<forall<a> (a) -> a>>
test6 : () -> ndet ((x : forall<a> (a) -> a) -> (forall<a> (a) -> a))
test6a : () -> ndet ((f : forall<a> (a) -> a) -> (forall<a> (a) -> a))
test7 : forall<a> () -> ndet ((f : forall<b> (b) -> b) -> ((a) -> a))
test8 : () -> ndet list<forall<a> (a) -> a>
test8a : () -> ndet list<forall<a> (a) -> a>
xauto : forall<a> (f : forall<b> (b) -> b) -> ((a) -> a)